diff --git a/Makefile.am b/Makefile.am
index 7a500ba..f5d4ecb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -41,7 +41,7 @@ distuninstallcheck_listfiles = find . -type f | \
 ACLOCAL_AMFLAGS = -I m4
 
 # subdirectories in the distribution
-SUBDIRS = lib src doc po tests util
+SUBDIRS = lib src @DOC_SUBDIR@ po tests util
 
 EXTRA_DIST = ChangeLog.README MAILING-LIST \
              msdos/ChangeLog msdos/config.h msdos/Makefile.DJ \
diff --git a/configure.ac b/configure.ac
index 873c3c9..15659f0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -90,6 +90,17 @@ ENABLE_DEBUG=$enableval, ENABLE_DEBUG=yes)
 test x"${ENABLE_DEBUG}" = xyes && AC_DEFINE([ENABLE_DEBUG], 1,
    [Define if you want the debug output support compiled in.])
 
+AC_ARG_ENABLE(doc,
+			  [  --disable-doc           "don't generate documentation"],
+ENABLE_DOC=$enableval, ENABLE_DOC=yes)
+if [ x"${ENABLE_DOC}" = xyes ] ; then
+	DOC_MAKEFILE=doc/Makefile
+	DOC_SUBDIR=doc
+	AC_SUBST(DOC_SUBDIR)
+else
+	DOC_MAKEFILE=
+fi
+
 dnl
 dnl Find the compiler
 dnl
@@ -552,7 +563,7 @@ AM_CONDITIONAL([IRI_IS_ENABLED], [test "X$iri" != "Xno"])
 dnl
 dnl Create output
 dnl
-AC_CONFIG_FILES([Makefile src/Makefile doc/Makefile util/Makefile
+AC_CONFIG_FILES([Makefile src/Makefile "$DOC_MAKEFILE" util/Makefile
                  po/Makefile.in tests/Makefile tests/WgetTest.pm
                  lib/Makefile])
 AC_CONFIG_HEADERS([src/config.h])
