diff --git a/src/video/fbcon/SDL_fbvideo.c b/src/video/fbcon/SDL_fbvideo.c
index 8daab7f..c077d82 100644
--- a/src/video/fbcon/SDL_fbvideo.c
+++ b/src/video/fbcon/SDL_fbvideo.c
@@ -1910,12 +1910,11 @@ static int FB_SetColors(_THIS, int firstcolor, int ncolors, SDL_Color *colors)
 static void FB_VideoQuit(_THIS)
 {
 	int i, j;
+	const char *dontClearPixels = SDL_getenv("SDL_FBCON_DONT_CLEAR");
 
 	if ( this->screen ) {
 		/* Clear screen and tell SDL not to free the pixels */
 
-		const char *dontClearPixels = SDL_getenv("SDL_FBCON_DONT_CLEAR");
-
 		/* If the framebuffer is not to be cleared, make sure that we won't
 		 * display the previous frame when disabling double buffering. */
 		if ( dontClearPixels && (this->screen->flags & SDL_DOUBLEBUF) && flip_page == 0 ) {
@@ -1971,7 +1970,7 @@ static void FB_VideoQuit(_THIS)
 		}
 
 		/* Restore the original video mode and palette */
-		if ( FB_InGraphicsMode(this) ) {
+		if ( FB_InGraphicsMode(this) && !dontClearPixels ) {
 			FB_RestorePalette(this);
 			ioctl(console_fd, FBIOPUT_VSCREENINFO, &saved_vinfo);
 		}
