diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3dc00e3..0174898 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,6 @@ set(MTP_VERSION_MINOR 0)
 set(MTP_VERSION_PATCH 0)
 
 find_package(Boost REQUIRED COMPONENTS thread system filesystem unit_test_framework)
-pkg_check_modules(DBUSCPP REQUIRED dbus-cpp)
 pkg_check_modules(GLOG REQUIRED libglog)
 
 set(
@@ -61,7 +60,6 @@ include_directories(
     include/
     libusbhost/include
     ${Boost_INCLUDE_DIRS}
-    ${DBUSCPP_INCLUDE_DIRS}
 )
 
 add_library(
@@ -72,7 +70,6 @@ add_library(
 target_link_libraries(
   mtpserver
   ${GLOG_LIBRARIES}
-  ${DBUSCPP_LIBRARIES}
 )
 
 set_target_properties(
@@ -94,10 +91,3 @@ install(
   RUNTIME DESTINATION lib
   PUBLIC_HEADER DESTINATION include/mtp
 )
-
-enable_testing()
-
-add_subdirectory(libusbhost)
-add_subdirectory(server)
-add_subdirectory(tests)
-add_subdirectory(po)
