diff -u libmikmod-3.1.11.org/configure libmikmod-3.1.11/configure
--- libmikmod-3.1.11.org/configure	2004-01-21 18:43:52.000000000 +0100
+++ libmikmod-3.1.11/configure	2012-10-02 02:37:49.000000000 +0200
@@ -2912,6 +2912,7 @@
 case $host_os in
 linux-gnu*) ;;
 linux*) host=`echo $host | sed 's/^\(.*-.*-linux\)\(.*\)$/\1-gnu\2/'`
+        host_os=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\3/'`
 esac
 
 case $host_os in
